main.go

```diff
@@ -97,50 +98,97 @@ func (a *KrakenApi) get_asset_pairs() map[string]interface{} {
 
 	// get list of asset-pairs and info about them (fees, etc)
 
-	var result2 map[string]interface{}
+	var data map[string]interface{}
 
 	req, err := http.NewRequest("POST", "https://api.kraken.com/0/public/AssetPairs", nil)
 
 	checkErr(err)
 	req.Header.Set("Api-key", a.key)
 	req.Header.Set("Api-Sign", a.secret)
 	resp, err := a.client.Do(req)
+	defer resp.Body.Close()
 	checkErr(err)
-	json.NewDecoder(resp.Body).Decode(&result2)
-	fmt.Println("DEBUG - ", result2)
-	return result2
+	json.NewDecoder(resp.Body).Decode(&data)
+	return data
+
+}
+
+func (a *KrakenApi) get_order_book(assetPair string) map[string]interface{} {
+	// given string "assetPair" of currency pair (example, BCHXBT)
+	// retrieve full orderbook
+	// returns array for asks and bids <price>, <volume>, <timestamp>
+
+	var data map[string]interface{}
+	params := url.Values{}
```

Comment:
not using this as more than a mapping currently but `urls` can hold onto a lot for us in POST data, starting this precedent now.